### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by DanielRrr**
At the moment, our codebase lacks documentation, this PR covers the library with explaining comments.

### CODE REVIEW

File: FFaCiL/EllipticCurve.lean [hunk 1]
```diff
@@ -143,6 +189,11 @@ def double (p : ProjectivePoint C) : ProjectivePoint C :=
 instance : Neg $ ProjectivePoint C where
   neg := fun ⟨x, y, z⟩ => ⟨x, -y, z⟩
 
+/--
+Addition in the Abelian group of projective points.
+The implementation is based on 
+Handbook of elliptic and hyperelliptic curve cryptography by Henri Cohen, et al., 13.2.1.b.
```
Comment (mpenciak) [inline FFaCiL/EllipticCurve.lean:195]: Is this from the handbook or the https://eprint.iacr.org/2015/1060.pdf paper?

File: FFaCiL/EllipticCurve.lean [hunk 2]
```diff
@@ -228,20 +309,42 @@ instance [PrimeField F] : Random $ AffinePoint C where
 
 end AffinePoint
 
--- variable {F K : Type _} [Field F] (C : Curve F) 
-
+/--
+Converts a normalised projective point to the affine one.
```
Comment (mpenciak) [inline FFaCiL/EllipticCurve.lean:313]:
```suggestion
Normalizes a projective point to the standard affine chart.
```

File: FFaCiL/GaloisField.lean [hunk 3]
```diff
@@ -143,13 +143,17 @@ instance [GaloisField L] [GaloisField K] [BEq K] [BEq L]
...
+/-- The type of residues
+-/
 inductive Residue where
   | zero
   | quadraticResidue
   | quadraticNonResidue
 deriving Repr
 
+/--
+Given `x : K`, legendreSymbol computes the Legendre symbol of `x`.
```
Comment (mpenciak) [inline FFaCiL/GaloisField.lean:155]: Maybe say that it checks whether `x` is a square.

File: FFaCiL/MSM.lean [hunk 4]
```diff
@@ -2,6 +2,9 @@ import FFaCiL.EllipticCurve
 import YatimaStdLib.ByteArray
```
Comment (mpenciak) [inline FFaCiL/MSM.lean:2]: Tomorrow afternoon I'll spend some time to explain what's going on in this and the Pasta files

**Review by mpenciak — CHANGES_REQUESTED** [review]

File: FFaCiL/EllipticCurve.lean (hunk 1 above)
Comment (DanielRrr) [reply FFaCiL/EllipticCurve.lean:195]: I can't remember, TBH

Comment (DanielRrr) [reply FFaCiL/EllipticCurve.lean:195]: Oh, it seems you're right

File: FFaCiL/MSM.lean [hunk 5]
```diff
@@ -53,7 +82,12 @@ def MSMInstance.evaluate (inst : MSMInstance C) (offset : Nat) : ProjectivePoint
     idx := idx - 1
   return (Nat.repeat ProjectivePoint.double offset) answer
 
+/--
+Pippenger's algorithm for recombining the separate evaluations from `MSMInstance.evaluate`.
+This implementation is parallelized and
```
Comment (DanielRrr) [inline FFaCiL/MSM.lean:87]: I think this sentence is unfinished

**Pull request merged**